code_samples/ai_actions/src/Command/AddMissingAltTextCommand.php

```diff
@@ -0,0 +1,158 @@
+    {
+        $this->addArgument('user', InputArgument::OPTIONAL, 'Login of the user executing the actions', 'admin');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $this->setUser($input->getArgument('user'));
+
+        $modifiedImages = $this->getModifiedImages();
+        $output->writeln(sprintf('Found %d modified image in the last 24h', $modifiedImages->getTotalCount()));
+
+        /** @var \Ibexa\Contracts\Core\Repository\Values\Content\Content $content */
+        foreach ($modifiedImages as $content) {
+            /** @var ?Value $value */
+            $value = $content->getFieldValue(self::IMAGE_FIELD_IDENTIFIER);
+
+            if ($value === null || !$this->shouldGenerateAltText($value)) {
+                $output->writeln(sprintf('Image %s has the image field empty or the alternative text is already specified. Skipping.', $content->getName()));
+                continue;
+            }
+
+            $contentUpdateStruct = $this->contentService->newContentUpdateStruct();
+            $value->alternativeText = $this->getSuggestedAltText($this->convertImageToBase64($value->uri), $content->getDefaultLanguageCode());
+            $contentUpdateStruct->setField(self::IMAGE_FIELD_IDENTIFIER, $value);
+
+            $updatedContent = $this->contentService->updateContent(
+                $this->contentService->createContentDraft($content->getContentInfo())->getVersionInfo(),
+                $contentUpdateStruct
+            );
+            $this->contentService->publishVersion($updatedContent->getVersionInfo());
+        }
+
+        return Command::SUCCESS;
+    }
+
+    private function getSuggestedAltText(string $imageEncodedInBase64, string $languageCode): string
+    {
+        $action = new GenerateAltTextAction(new Image([$imageEncodedInBase64]));
+
+        $action->setRuntimeContext(new RuntimeContext(['languageCode' => $languageCode]));
+        $action->setActionContext(
+            new ActionContext(
+                new ActionConfigurationOptions(['default_locale_fallback' => 'en']), // System context
+                new ActionConfigurationOptions(['max_lenght' => 100]), // Action Type options
+                new ActionConfigurationOptions( // Action Handler options
+                    [
+                        'prompt' => 'Generate the alt text for this image in less than 100 characters.',
+                        'temperature' => 0.7,
+                        'max_tokens' => 4096,
+                        'model' => 'gpt-4o-mini',
+                    ]
+                )
+            )
+        );
+
+        $output = $this->actionService->execute($action)->getOutput();
+
+        assert($output instanceof Text);
+
+        return $output->getText();
+    }
+
+    private function convertImageToBase64(?string $uri): string
+    {
+        $file = file_get_contents($this->projectDir . \DIRECTORY_SEPARATOR . 'public' . \DIRECTORY_SEPARATOR . $uri);
```

Comment:
Shouldn't this use the binary/IO Handler or similar abstract layer to be compatible with DFS or other cluster storage?

Reply:
Thank you for the suggestion! Ok for you to improve this in a follow-up? I agree it would be best to show a best practise here.

I think https://github.com/ibexa/core/blob/4.6/src/lib/IO/IOBinarydataHandler.php is what I might be looking for, but I will confirm that